src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrder.java

```diff
@@ -0,0 +1,165 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.TextBlock;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+
+ */
+@StatelessCheck
+public class JavadocParamOrder extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    @Override
+    public final int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.ANNOTATION_FIELD_DEF,
+            TokenTypes.RECORD_DEF,
+            TokenTypes.COMPACT_CTOR_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final List<DetailAST> methodParams = getParameters(ast);
+        final List<DetailAST> methodTypeParams = CheckUtil
+                .getTypeParameters(ast);
+
+        final FileContents contents = getFileContents();
+        final int lineNo = ast.getLineNo();
+        final TextBlock textBlock = contents.getJavadocBefore(lineNo);
+
+        final List<JavadocTag> javadocParams = getJavadocTags(textBlock);
+
+        method1(methodParams, methodTypeParams, javadocParams);
+    }
+
+    /**
+     * Checks a set of tags for matching parameters.
+     *
+     *  @param methodParams list of method parameters
+     * @param params list of method type parameters
+     * @param javadocParams list of javadoc parameters
+     */
+    private void method1(final List<DetailAST> methodParams,
+                                final List<DetailAST> params,
+                         final List<JavadocTag> javadocParams) {
+
+        final int totalMethodParamSize = methodParams.size() + params.size();
+        //Merge both lists
+        params.addAll(methodParams);
+
+        final ListIterator<JavadocTag> javadocTag = javadocParams.listIterator();
+        final ListIterator<DetailAST> methodTag = params.listIterator();
+        while (javadocTag.hasNext() && methodTag.hasNext()) {
```

Comment:
Size of both should be same.

Reply:
should be same but if not like u said below, there could be other tags as well not only params